### PR TITLE
WINDUP-482: Throw an exception in the case of rulesets with duplicate id...

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-482

### CODE REVIEW

File: config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java [hunk 1]
```diff
@@ -78,6 +80,20 @@ private void printRulePhases(List<WindupRuleProvider> allProviders)
...
 
+    private void checkForDuplicates(List<WindupRuleProvider> providers)
+    {
+        // using a map so that we can easily pull out the previous value later (in the case of a duplicate)
+        Set<WindupRuleProvider> dupeSet = new HashSet<>(providers.size());
+        for (WindupRuleProvider provider : providers)
+        {
+            if (dupeSet.contains(provider))
+            {
+                throw new WindupException("Found two providers with the same id: " + provider.getID());
```
Comment (lincolnthree) [inline config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java]: I wonder if this should also list the duplicate instances / types?

Comment (jsight) [inline config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java]: Yeah, possibly. Although I expect that most of the time the class will be WindupRuleProviderBuilder.

Comment (lincolnthree) [inline config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java]: True.

File: rules-java-archives/addon/pom.xml [hunk 2]
```diff
@@ -11,7 +11,7 @@
     </parent>
 
     <artifactId>rules-java-archives</artifactId>
-    <name>Windup Ruleset: Archive Identification - Addon</name>
```
Comment (lincolnthree) [inline rules-java-archives/addon/pom.xml]: Thank you.

**Pull request merged**